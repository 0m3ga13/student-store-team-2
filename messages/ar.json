{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك"
    }
}
