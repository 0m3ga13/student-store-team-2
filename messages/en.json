{
    "Index": {
        "Title": "MiniStore",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Free delivery": "Free delivery",
        "feature descriprion 1": "You can receive your order without incurring any additional shipping costs",
        "qualite guarantee": "qualite guarantee",
        "feature descriprion 2": "Our products or services meet high standards and perform as promised.",
        "daily offers": "daily offers",
        "feature descriprion 3": "Exclusive discounts and promotions on a daily basis",
        "100% secure payement": "100% secure payement",
        "feature descriprion 4": "Your financial information is protected and your transactions are safe from unauthorized access",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "my stuff":"my stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "electronics": "electronics",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product."
    }
}
