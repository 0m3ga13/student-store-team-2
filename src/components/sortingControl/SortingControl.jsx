"use client";
import { useTranslations } from "next-intl";
import React from "react";

const SortingControl = () => {
    const t = useTranslations("Index");
    return (
        <div className='font-lato font-semibold text-lg text-titleContent flex justify-center w-fit sm:w-full lg:w-3/4 sm:h-16'>
            <div className='md:flex justify-center items-center text-center sm:px-4  mr-2 mb-4 w-1/2 hidden'>
                <div className=' hover:underline mx-2'>{t("All")}</div>
                <div className=' hover:underline mx-2'>{t("Pending")}</div>
                <div className=' hover:underline mx-2'>{t("Active")}</div>
                <div className=' hover:underline mx-2'>{t("Completed")}</div>
            </div>
            <div className='ml-1/10 mr-2 mb-4 sort-list md:hidden flex justify-between px-4 items-center'>
                <label className='sort-label pr-2  hover:underline hidden'>
                    {t("Filter")}
                </label>
                <select className='font-lato text-secondaryGray text-lg sort-select h-14 w-32 border-2 rounded-lg'>
                    <option
                        value='All'
                        className='text-secondaryGray text-center hover:bg-accent hover:text-white h-16 w-44'
                    >
                        {t("All")}
                    </option>
                    <option
                        value='Pending'
                        className='text-secondaryGray text-center hover:bg-accent hover:text-white h-16 w-44'
                    >
                        {t("Pending")}
                    </option>
                    <option
                        value='Active'
                        className='text-secondaryGray text-center hover:bg-accent hover:text-white h-16 w-44'
                    >
                        {t("Active")}
                    </option>
                    <option
                        value='Completed'
                        className='text-secondaryGray text-center hover:bg-accent hover:text-white h-16 w-44'
                    >
                        {t("Completed")}
                    </option>
                </select>
            </div>
            <div className='ml-1/10 mb-4 sort-list flex px-4 items-center'>
                <label className='sort-label pr-2 hidden hover:underline'>
                    {t("Sort")}
                </label>
                <select className='font-lato text-secondaryGray text-lg sort-select h-14 w-36 border-2 rounded-lg'>
                    <option
                        value='newest'
                        className='text-secondaryGray text-center hover:bg-accent hover:text-white h-16 w-44'
                    >
                        {t("Newest")}
                    </option>
                    <option
                        value='oldest'
                        className='text-secondaryGray text-center hover:bg-accent hover:text-white h-16 w-44'
                    >
                        {t("Oldest")}
                    </option>
                </select>
            </div>
        </div>
    );
};

export default SortingControl;
