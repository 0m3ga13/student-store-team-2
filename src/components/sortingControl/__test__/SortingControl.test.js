import { NextIntlClientProvider } from "next-intl";
import renderer from "react-test-renderer";

import SortingControl from "../SortingControl";
import messages from "../../../../messages/en.json";

it("renders correctly", () => {
    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <SortingControl />
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
