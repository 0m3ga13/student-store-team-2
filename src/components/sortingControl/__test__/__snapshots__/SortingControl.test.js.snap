// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="font-lato font-semibold text-xl text-titleContent sm:flex justify-center w-[940px] h-16 hidden sm:block"
>
  <div
    className="flex justify-between items-center text-center space-x-4 mb-4  w-[630px] "
  >
    <div
      className=" hover:underline"
    >
      All
    </div>
    <div
      className=" hover:underline"
    >
      Awaiting Pickup
    </div>
    <div
      className=" hover:underline"
    >
      Active
    </div>
    <div
      className=" hover:underline"
    >
      Completed
    </div>
  </div>
  <div
    className="ml-48 mb-4 sort-list flex items-center "
  >
    <label
      className="sort-label pr-2  hover:underline"
    >
      Sort:
    </label>
    <select
      className="font-lato text-secondaryGray text-xl sort-select px-2 h-16 w-44 border-2 rounded-lg"
    >
      <option
        className="text-secondaryGray hover:bg-accent hover:text-white h-16 w-44"
        value="newest"
      >
        Newest
      </option>
      <option
        className="text-secondaryGray hover:bg-accent hover:text-white h-16 w-44"
        value="oldest"
      >
        Oldest
      </option>
    </select>
  </div>
</div>
`;
